src/ui/box.js

```diff
@@ -36,18 +36,27 @@ class Renderer {
       CSSCore.addClass(window.document.getElementsByTagName('html')[0], 'auth0-lock-html');
     }
     // eslint-disable-next-line
-    const component = ReactDOM.render(<Container {...props} />, container);
+    const root = this.modals[containerId] ? this.modals[containerId].root : createRoot(container);
 
-    if (isModal) {
-      this.modals[containerId] = component;
+    if (!this.modals[containerId]) {
+      this.modals[containerId] = { root };
     }
 
-    return component;
+    root.render(
+      <Container
+        {...props}
+        ref={component => {
+          if (component && isModal) {
+            this.modals[containerId].component = component;
+          }
+        }}
+      />
+    );
```

Comment:
Since creation of React root and rendering are now separate, `root` should be stored for later. Since Lock depends on having an instance of the created component (mainly so that it can close it again), I use a callback ref to grab it and store `component` as before, rather than rock the boat too much and rewrite how closing is done.

https://reactjs.org/docs/react-dom.html#:~:text=render()%20currently,the%20root%20element.